ui-v2/app/styles/components/icons/index.scss

```diff
@@ -1,15 +1,35 @@
-%pseudo-icon {
-  width: 1em;
-  height: 1em;
-  position: absolute;
-  top: 50%;
-  margin-top: -0.6em;
+/*TODO: The old pseudo-icon was to specific */
+/* make a temporary one with the -- prefix */
+/* to make it more reusable temporarily */
+%--pseudo-icon {
   display: block;
   content: '';
+  visibility: visible;
+  position: absolute;
+  top: 50%;
   background-repeat: no-repeat;
   background-position: center center;
+}
+%pseudo-icon-bg-img {
+  @extend %--pseudo-icon;
+  background-size: contain;
+  background-color: transparent;
+}
+%pseudo-icon-css {
+  @extend %--pseudo-icon;
+  width: 1em;
+  height: 1em;
+  margin-top: -0.6em;
   background-color: currentColor;
-  visibility: visible;
+}
+%pseudo-icon {
+  @extend %pseudo-icon-css;
+}
+%with-external-source-icon {
+  background-repeat: no-repeat;
+  background-size: contain;
+  width: 18px;
+  height: 18px;
 }
```

Comment:
Whilst doing this I realised my `%pseudo-icon` was doing too much. So for the moment I've split it out without changing where I'm actually using it. I'll probably sift through and change all the places I'm using `%pseudo-icon` to `%pseudo-icon-css` in a separate PR.